File: SistemaBancario/src/ContaEspecial.cpp
#include "ContaEspecial.h"
ContaEspecial::ContaEspecial(int numero, string cliente, double limite) : ContaSimples(numero, cliente){
	this->limite = limite;
}
ContaEspecial::~ContaEspecial(){
	//std::cout << "'destruindo' especial\n";
}

bool ContaEspecial::efetuaSaque(double valor){
	if(this->saldo + this->limite >= valor){
		this->saldo -= valor;
		return true;
	}
	return false;
}

double ContaEspecial::getSaldo(){
	return this->saldo + this->limite;
}

double ContaEspecial::getLimite(){
	return this->limite;
}

File: SistemaBancario_java/src/ContaEspecial.java

public class ContaEspecial extends ContaSimples{
	
	private double limite;
	
	public ContaEspecial(){}
	
	public ContaEspecial(int numero, String cliente, double limite){
		super(numero,cliente);
		this.limite = limite;
	}
	
	
	boolean efetuaSaque(double valor){
		if(super.getSaldo() + this.limite >= valor){
			setSaldo(super.getSaldo() - valor);
			return true;
		}
		return false;
	}

	double getSaldo(){
		return super.getSaldo() + this.limite;
	}

	double getLimite(){
		return this.limite;
	}


}
